editoast/src/views/v2/scenario.rs

```diff
@@ -0,0 +1,629 @@
+        timetable_id,
+    })
+    .await?;
+
+    // Create Scenario
+    let scenario = scenario.study_id(study_id).create(conn).await?;
+
+    // Update study last_modification field
+    study.clone().update_last_modified(conn).await?;
+
+    // Update project last_modification field
+    project.update_last_modified(conn).await?;
+
+    let scenarios_with_details = scenario.with_details_conn(conn).await?;
+
+    let scenarios_response = ScenarioResponseV2::new(scenarios_with_details, project, study);
+    Ok(Json(scenarios_response))
+}
+
+/// Delete a scenario
+#[utoipa::path(
+    tag = "scenariosv2",
+    params(ProjectIdParam, StudyIdParam, ScenarioIdParam),
+    responses(
+        (status = 204, description = "The scenario was deleted successfully"),
+        (status = 404, body = InternalError, description = "The requested scenario was not found"),
+    )
+)]
+#[delete("")]
+async fn delete(path: Path<(i64, i64, i64)>, db_pool: Data<DbPool>) -> Result<HttpResponse> {
+    use crate::modelsv2::DeleteStatic;
+    let (project_id, study_id, scenario_id) = path.into_inner();
+
+    // Check if the project and the study exist
+    let (mut project, study) = check_project_study(db_pool.clone(), project_id, study_id)
+        .await
+        .unwrap();
+
+    let conn = &mut db_pool.get().await?;
+
+    // Delete scenario
+    ScenarioModel::delete_static_or_fail(conn, scenario_id, || ScenarioError::NotFound {
+        scenario_id,
+    })
+    .await?;
+
+    // Update project last_modification field
+    project.update_last_modified(conn).await?;
+
+    // Update study last_modification field
+    study.clone().update_last_modified(conn).await?;
+
+    Ok(HttpResponse::NoContent().finish())
+}
+
+/// This structure is used by the patch endpoint to patch a study
+#[derive(Serialize, Deserialize, Derivative, ToSchema)]
+#[derivative(Default)]
+struct ScenarioPatchFormV2 {
+    pub name: Option<String>,
+    pub description: Option<String>,
+    pub tags: Option<Vec<Option<String>>>,
+    pub infra_id: Option<i64>,
+}
+
+impl From<ScenarioPatchFormV2> for <ScenarioModel as crate::modelsv2::Model>::Changeset {
+    fn from(scenario: ScenarioPatchFormV2) -> Self {
+        Self {
+            name: scenario.name,
+            description: scenario.description,
+            tags: scenario.tags,
+            infra_id: scenario.infra_id,
+            ..Default::default()
+        }
+    }
+}
+
+/// Update a scenario
+#[utoipa::path(
+    tag = "scenariosv2",
+    params(ProjectIdParam, StudyIdParam, ScenarioIdParam),
+    request_body = ScenarioPatchFormV2,
+    responses(
+        (status = 204, body = ScenarioResponseV2, description = "The scenario was updated successfully"),
+        (status = 404, body = InternalError, description = "The requested scenario was not found"),
+    )
+)]
+#[patch("")]
+async fn patch(
+    data: Json<ScenarioPatchFormV2>,
+    path: Path<(i64, i64, i64)>,
```

Comment:
```suggestion
    path: Path<ScenarioPathParam>,
```

which would be a struct with these three fields + IntoParams. Path<> implements Deref so you might not have to .into_inner() as well.

Reply:
i tried but i can't implement Deref for Path<> it a external crate